src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java

```diff
@@ -3,30 +3,100 @@
     }
 
     @Override
-    public Comment create(Comment entity) {
-        return null;
+    public Comment create(Comment comment) {
+        log.info("Calling a create() method of CommentDaoImpl class");
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            session.beginTransaction();
+            session.save(comment);
+            log.info("Attempt to save comment " + comment + " in db.");
+            session.getTransaction().commit();
+            return comment;
+        } catch (Exception e) {
+            if (session.getTransaction() != null) {
+                session.getTransaction().rollback();
+            }
+            throw new RuntimeException("Can't insert Content entity");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Comment get(Long id) {
-        return null;
+        log.info("Calling a get() method of CommentDaoImpl class");
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            session.beginTransaction();
+            Comment comment = (Comment) session.get(Comment.class, id);
+            log.info("Attempt to retrieve comment " + comment + " from db.");
+            session.flush();
+            return comment;
+        } catch (Exception e) {
+            if (session.getTransaction() != null) {
+                session.getTransaction().rollback();
+            }
+            throw new RuntimeException("Can't insert comment entity. ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Comment> getAll() {
-        return null;
+        log.info("Calling a getAll() method of CommentDaoImpl class");
+        try (Session session = factory.openSession()) {
+            CriteriaQuery<Comment> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Comment.class);
+            criteriaQuery.from(Comment.class);
+            log.info("Attempt to retrieve collection of comments from db.");
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Error retrieving all comments. ", e);
+        }
     }
 
     @Override
-    public void remove(Comment entity) {
-
+    public void remove(Comment comment) {
+        log.info("Calling a remove() method of CommentDaoImpl class");
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            session.beginTransaction();
+            session.remove(comment);
+            log.info("Attempt to remove comment " + comment + " from db");
+            session.getTransaction().commit();
```

Comment:
You create a Transaction object at line 83 - maybe it's better to use it here? I think Bohdan said in today's lecture that it is better to use transaction.commit() because we could have many transactions in one session, so when we use session.getTransaction(), we might get the wrong transaction. Something like that.